pkg/api/v1beta3/types.go

```diff
@@ -995,6 +1006,16 @@ type Binding struct {
 	Target ObjectReference `json:"target" description:"an object to bind to"`
 }
 
+// DeleteOptions may be provided when deleting an API object
+type DeleteOptions struct {
+	TypeMeta `json:",inline"`
+
+	// Optional duration in seconds before the object should be deleted. Value must be non-negative integer.
+	// The value zero indicates delete immediately. If this value is nil, the default grace period for the
+	// specified type will be used.
+	GracePeriod *int64 `json:"gracePeriod" description:"the duration in seconds to wait before deleting this object; defaults to a per object value if not specified; zero means delete immediately"`
```

Comment:
Yes.  It's specified on deletion timestamp but not on the delete options object.  I'll add that (you can shorten but not lengthen grace period).


> On Mar 17, 2015, at 12:52 AM, Brian Grant <notifications@github.com> wrote:
> 
> In pkg/api/v1beta3/types.go:
> 
> > @@ -995,6 +1006,16 @@ type Binding struct {
> >  	Target ObjectReference `json:"target" description:"an object to bind to"`
> >  }
> >  
> > +// DeleteOptions may be provided when deleting an API object
> > +type DeleteOptions struct {
> > +	TypeMeta `json:",inline"`
> > +
> > +	// Optional duration in seconds before the object should be deleted. Value must be non-negative integer.
> > +	// The value zero indicates delete immediately. If this value is nil, the default grace period for the
> > +	// specified type will be used.
> > +	GracePeriod *int64 `json:"gracePeriod" description:"the duration in seconds to wait before deleting this object; defaults to a per object value if not specified; zero means delete immediately"`
> Once specified, is it possible to change deletionTimestamp (to make it nearer or farther) with subsequent DELETEs? Allowing the effective grace period to be shortened could be useful.
> 
> —
> Reply to this email directly or view it on GitHub.
>